src/screens/WalletConnect/CallRequest/SignatureRequestContent.js

```diff
@@ -55,11 +60,53 @@ function SignatureRequestContent({ request, onConfirm, onReject }: Props) {
   const { t } = useTranslation();
   const configs = useChainsConfig();
   const activeAccount = useActiveAccount();
+  const colors = useThemeColors();
   const isArchanovaAccountDeployed = useRootSelector(isArchanovaAccountDeployedSelector);
 
-  const { title, iconUrl, chain } = getViewData(request);
+  const { title, iconUrl, chain, method, params } = getViewData(request);
   const config = configs[chain];
 
+  const isSignTypedData = method === ETH_SIGN_TYPED_DATA || method === ETH_SIGN_TYPED_DATA_V4;
+
+  const { message, primaryType } =
+    isSignTypedData && !isEmpty(params) ? JSON.parse(params[1]) : { message: null, primaryType: null };
```

Comment:
`JSON.parse(params[1])` will throw an error if that index doesn't exist. Could we check for this or fail gracefully?

Reply:
Condition updated.